flags/models.py

```diff
@@ -13,3 +22,73 @@ class Meta:
     def __str__(self):
         return "{name} is enabled when {condition} is {value}".format(
             name=self.name, condition=self.condition, value=self.value)
+
+
+class FlaggablePageMixin(models.Model):
+    """ This is a model mixin class can be be mixed-in along side the Wagtail
+    Page model (or a subclass thereof) to add the viewing of the latest draft
+    revision instead of the published revision based on feature flag state.
+    It is an alternative to inheriting directly from FlaggablePage. """
+
+    feature_flag_name = models.CharField(
+        verbose_name="Feature flag",
+        max_length=64,
+        blank=True,
+        help_text=("The selected feature flag will control whether the draft "
+                   "version of this page is visible live for the feature "
+                   "flag's conditions")
+    )
+    show_draft_with_feature_flag = models.BooleanField(
+        verbose_name="Show draft",
+        default=False,
+        help_text=("Show the latest draft of this page live regardless of "
+                   "published status, when the selected feature flag is "
+                   "enabled.")
+    )
+
+    flag_panels = [
+        FieldPanel('show_draft_with_feature_flag', 'Feature flag enabled'),
+        FlagChooserPanel('feature_flag_name'),
+    ]
+
+    settings_panels = [
+        MultiFieldPanel(flag_panels, 'Feature flagged drafts'),
+    ]
+
+    class Meta:
+        abstract = True
+
+    def serve_flaggable(self, request, *args, **kwargs):
+        latest_revision = self.get_latest_revision_as_page()
+        if (flag_enabled(latest_revision.feature_flag_name, request=request)
+                and latest_revision.show_draft_with_feature_flag):
+            return Page.serve(latest_revision, request, *args, **kwargs)
+
+        return Page.serve(self, request, *args, **kwargs)
```

Comment:
Is there any way of using this mixin with multiple levels of inheritance, for example:

```py
class MyBasePageClass(Page):
    ...

class MyFlaggableBasePageClass(FlaggablePageMixin, MyBasePageClass):
    # because I don't want all page types to be flaggable
    # but if this calls serve_flaggable, it'll call Page.serve, MyBasePageClass.serve
```

How would you define the latter class while avoiding breaking things when `Page.serve` is called instead of `MyBasePageClass.serve`?